apps/prairielearn/src/pages/instructorGradebook/components/InstructorGradebookTable.tsx

```diff
@@ -0,0 +1,544 @@
+        columnHelper.group({
+          id: `group_${setId}`,
+          header: assessmentsBySet.headingById.get(setId) ?? 'Unknown',
+          columns: assessments.map((assessment) =>
+            columnHelper.accessor(
+              (row) => {
+                const data = row.scores[assessment.assessment_id];
+                // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+                return data ? data.score_perc : null;
+              },
+              {
+                id: `a${assessment.assessment_id}`,
+                minSize: 80,
+                maxSize: 500,
+                meta: {
+                  label: assessment.label,
+                  autoSize: true,
+                },
+                header: () => (
+                  <a href={`${urlPrefix}/assessment/${assessment.assessment_id}`}>
+                    <span
+                      class={clsx('badge', `color-${assessment.color}`)}
+                      title={assessment.label}
+                    >
+                      {assessment.label}
+                    </span>
+                  </a>
+                ),
+                cell: (info) => {
+                  const score = info.getValue();
+                  const row = info.row.original;
+                  const assessmentData = row.scores[assessment.assessment_id];
+
+                  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+                  if (score == null || !assessmentData?.assessment_instance_id) {
+                    return '—';
+                  }
+
+                  return (
+                    <span class="text-nowrap">
+                      <a
+                        href={`${urlPrefix}/assessment_instance/${assessmentData.assessment_instance_id}`}
+                      >
+                        {Math.floor(score)}%
+                      </a>
+                      <EditScoreButton
+                        assessmentInstanceId={assessmentData.assessment_instance_id}
+                        courseInstanceId={courseInstanceId}
+                        currentScore={score}
+                        otherUsers={assessmentData.uid_other_users_group}
+                        csrfToken={csrfToken}
+                      />
+                    </span>
+                  );
+                },
+                filterFn: numericColumnFilterFn,
+              },
+            ),
+          ),
+        }),
+      ),
+    ],
+    [assessmentsBySet.groups, assessmentsBySet.headingById, urlPrefix, courseInstanceId, csrfToken],
+  );
+
+  // Extract only leaf column IDs (exclude group columns)
+  const allColumnIds = extractLeafColumnIds(columns);
+
+  // Set default visibility: hide name, UIN, role, and enrollment status columns by default
+  const defaultColumnVisibility = Object.fromEntries(
+    allColumnIds.map((id) => {
+      if (['user_name', 'uin', 'role', 'enrollment_status'].includes(id)) {
+        return [id, false];
```

Comment:
I think we should show `user_name` by default by not freeze it. This matches the current functionality and also makes it feel generally more useable for me. The current gradebook also shows UIN by default but I don't think that's necessary.